sktime/forecasting/base/adapters/_neuralforecast.py

```diff
@@ -179,6 +181,16 @@ def _fit(
         if not fh.is_all_out_of_sample(cutoff=self.cutoff):
             raise NotImplementedError("in-sample prediction is currently not supported")
 
+        if self.freq == "auto":
+            if fh.freq:
+                # interpret freq from ForecastingHorizon
+                self.freq = fh.freq
+            else:
+                # when freq is not interpreted from ForecastingHorizon
+                raise ValueError(
+                    "Could not interpret freq, try passing freq in model initialization"
```

Comment:
It's better if you edit the message to fail with model name. Something like `self.__class__.__name__` may work.